### PR TITLE
Cleanup my Community Layout

### PR DESCRIPTION
**Pull request by mrsendyyk**
## Description

Cleanup my code

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by mrsendyyk — COMMENTED** [review]
Ready for review

**Review by drashna — APPROVED** [review]

File: layouts/community/numpad_5x4/layout.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[{a:7},"","","",""],
```
Comment (noroadsleft) [inline layouts/community/numpad_5x4/layout.json:1]: You don't actually need this file here at all.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: layouts/community/numpad_5x4/mrsendyyk/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Sendy YK's Numpad 5x4 Layout and Keymap
+
+[https://mr.sendyyk.com](https://mr.sendyyk.com)
+
+## Numpad 5x4 Layout
+
+![LAYOUT_numpad_5x4](https://raw.githubusercontent.com/mrsendyyk/my_qmk/master/soyuz/assets/layout-numpad-5x4.png)
```
Comment (noroadsleft) [inline layouts/community/numpad_5x4/mrsendyyk/readme.md:7]: Note: These numbers are meant to describe the matrix wiring for the board being used with the keymap. These numbers are accurate for the Soyuz, but some boards with this layout wire things differently.

File: layouts/community/numpad_5x4/layout.json (hunk 1 above)
Comment (mrsendyyk) [reply layouts/community/numpad_5x4/layout.json:1]: Do you mean, i'm safe to delete  `layouts/community/numpad_5x4/layout.json` file @noroadsleft?

File: layouts/community/numpad_5x4/mrsendyyk/readme.md (hunk 2 above)
Comment (mrsendyyk) [reply layouts/community/numpad_5x4/mrsendyyk/readme.md:7]: Yes

**Review by zvecr — CHANGES_REQUESTED** [review]
This kinda "conflicts" with at least one of your existing keymaps `keyboards/ai03/soyuz/keymaps/mrsendyyk/keymap.c` as it shares the same name on a keyboard that also supports the community layout. Using the community layout would allow you to delete any keyboard specific keymaps. The alternative would be to name this or existing ones differently.

**Comment by mrsendyyk** [comment]
> 
> 
> This kinda "conflicts" with at least one of your existing keymaps `keyboards/ai03/soyuz/keymaps/mrsendyyk/keymap.c` as it shares the same name on a keyboard that also supports the community layout. Using the community layout would allow you to delete any keyboard specific keymaps. The alternative would be to name this or existing ones differently.



> 
> 
> This kinda "conflicts" with at least one of your existing keymaps `keyboards/ai03/soyuz/keymaps/mrsendyyk/keymap.c` as it shares the same name on a keyboard that also supports the community layout. Using the community layout would allow you to delete any keyboard specific keymaps. The alternative would be to name this or existing ones differently.

Done.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
General comment: at this point, you could actually remove the `mrsendyyk` folders from `dz60` and `ai03/soyuz` completely, rename both your folders under `layouts/community` back to `mrsendyyk`, and still generate your firmware by running `make dz60:mrsendyyk` or `make ai03/soyuz:mrsendyyk`:

```shell
QMK Firmware 0.10.20
Making dz60 with keymap mrsendyyk

avr-gcc.exe (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: keyboards/dz60/dz60.c                                                                    [OK]
Compiling: layouts/community/60_ansi_arrow/mrsendyyk/keymap.c                                       [OK]
...
```

```shell
QMK Firmware 0.10.20
Making ai03/soyuz with keymap mrsendyyk

avr-gcc.exe (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: keyboards/ai03/soyuz/soyuz.c                                                             [OK]
Compiling: layouts/community/numpad_5x4/mrsendyyk/keymap.c                                          [OK]
...
```

This has the advantage that you don't have to maintain multiple copies of the same files. It's up to you though. Let us know how you want to proceed.

**Comment by mrsendyyk** [comment]
> 
> 
> General comment: at this point, you could actually remove the `mrsendyyk` folders from `dz60` and `ai03/soyuz` completely, rename both your folders under `layouts/community` back to `mrsendyyk`, and still generate your firmware by running `make dz60:mrsendyyk` or `make ai03/soyuz:mrsendyyk`:
> 
> ```shell
> QMK Firmware 0.10.20
> Making dz60 with keymap mrsendyyk
> 
> avr-gcc.exe (GCC) 8.3.0
> Copyright (C) 2018 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> 
> Compiling: keyboards/dz60/dz60.c                                                                    [OK]
> Compiling: layouts/community/60_ansi_arrow/mrsendyyk/keymap.c                                       [OK]
> ...
> ```
> 
> ```shell
> QMK Firmware 0.10.20
> Making ai03/soyuz with keymap mrsendyyk
> 
> avr-gcc.exe (GCC) 8.3.0
> Copyright (C) 2018 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> 
> Compiling: keyboards/ai03/soyuz/soyuz.c                                                             [OK]
> Compiling: layouts/community/numpad_5x4/mrsendyyk/keymap.c                                          [OK]
> ...
> ```
> 
> This has the advantage that you don't have to maintain multiple copies of the same files. It's up to you though. Let us know how you want to proceed.

Did you mean it's looks like i have universal keymap based on layout @noroadsleft? So I just have to choose keyboards folder to make firmware files right?

**Comment by noroadsleft** [comment]
> Did you mean it's looks like i have universal keymap based on layout @noroadsleft? So I just have to choose keyboards folder to make firmware files right?

Exactly this. The whole point of the Community Layouts feature is that you write one keymap that can be flashed to any keyboard that supports the physical layout. `make` will look in the keyboard's `keymaps` folder for the keymap you specify, and when it doesn't find the keymap there, it checks the `layouts` folder and uses the keymap found there.

If you have a board that supports multiple community layouts (like the DZ60), and you have a keymap for more than one of the supported layouts, you can use the `FORCE_LAYOUT` flag to specify which layout to compile.

Example:

Say you purchase another DZ60, and build it with the [60_ansi_split_bs_rshift](https://user-images.githubusercontent.com/18669334/94348078-54e7d400-ffee-11ea-8e2d-bcfa49e8d4f6.png) layout, and add a `layouts/community/60_ansi_split_bs_rshift/mrsendyyk/` keymap.

You would compile your current `60_ansi_arrow` keymap by running `make dz60:mrsendyyk FORCE_LAYOUT=60_ansi_arrow`, and the new layout by running `make dz60:mrsendyyk FORCE_LAYOUT=60_ansi_split_bs_rshift`.

**Comment by mrsendyyk** [comment]
> 
> 
> > Did you mean it's looks like i have universal keymap based on layout @noroadsleft? So I just have to choose keyboards folder to make firmware files right?
> 
> Exactly this. The whole point of the Community Layouts feature is that you write one keymap that can be flashed to any keyboard that supports the physical layout. `make` will look in the keyboard's `keymaps` folder for the keymap you specify, and when it doesn't find the keymap there, it checks the `layouts` folder and uses the keymap found there.
> 
> If you have a board that supports multiple community layouts (like the DZ60), and you have a keymap for more than one of the supported layouts, you can use the `FORCE_LAYOUT` flag to specify which layout to compile.
> 
> Example:
> 
> Say you purchase another DZ60, and build it with the [60_ansi_split_bs_rshift](https://user-images.githubusercontent.com/18669334/94348078-54e7d400-ffee-11ea-8e2d-bcfa49e8d4f6.png) layout, and add a `layouts/community/60_ansi_split_bs_rshift/mrsendyyk/` keymap.
> 
> You would compile your current `60_ansi_arrow` keymap by running `make dz60:mrsendyyk FORCE_LAYOUT=60_ansi_arrow`, and the new layout by running `make dz60:mrsendyyk FORCE_LAYOUT=60_ansi_split_bs_rshift`.

Done @noroadsleft

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Looks good, thanks!

**Pull request merged**